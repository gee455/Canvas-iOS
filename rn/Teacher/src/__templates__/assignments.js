//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
