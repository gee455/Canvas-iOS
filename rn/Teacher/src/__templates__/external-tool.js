//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const externalTool: Template<AddressBookResult> = template({
  id: '1',
  name: 'LTI FTW',
  url: 'https://canvas.instructure.com/lti',
})

export const ltiLaunchDefinition: Template<LtiLaunchDefinition> = template({
  definition_id: 42,
  placements: {
    course_navigation: {
      url: 'https://rollcall.instructure.com/launchy-launch',
    },
  },
})
