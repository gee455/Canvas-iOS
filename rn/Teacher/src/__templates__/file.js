//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const usageRights: Template<UsageRights> = template({
  legal_copyright: '',
  use_justification: 'own_copyright',
})

export const file: Template<File> = template({
  id: '111',
  display_name: 'Book Report',
  url: 'http://canvaslms.com/bookreport',
  size: 488212,
  thumbnail_url: 'http://fillmurray.com/322/200',
  mime_class: 'document',
  preview_url: '/',
  filename: '/bookreport',
  parent_folder_id: null,
  locked: false,
  hidden: false,
  unlock_at: null,
  lock_at: null,
  usage_rights: usageRights(),
  'content-type': 'application/document',
})
