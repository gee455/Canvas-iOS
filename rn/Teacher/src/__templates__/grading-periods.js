//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const gradingPeriod: Template<GradingPeriod> = template({
  id: '1023',
  title: 'First Block',
  start_date: '2014-01-07T15:04:00Z',
  end_date: '2014-05-07T17:07:00Z',
  close_date: '2014-06-07T17:07:00Z',
  weight: 33.33,
})
