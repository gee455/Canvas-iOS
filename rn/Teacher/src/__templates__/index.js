//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
