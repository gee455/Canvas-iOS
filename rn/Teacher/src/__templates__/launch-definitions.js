//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const launchDefinitionGlobalNavigationItem: Template<ExternalToolLaunchDefinitionGlobalNavigationItem> = template({
  message_type: 'basic-lti-launch-request',
  url: 'https://mobiledev.beta.instructuremedia.com/lti/launch?custom_gauge_launch_type=global_nav',
  title: 'Gauge',
})

export const launchDefinition: Template<ExternalToolLaunchDefinition> = template({
  definition_type: 'ContextExternalTool',
  definition_id: '360860',
  name: 'Gauge',
  description: 'Tools for Education',
  domain: 'gauge.instructure.com',
  placements: {
    global_navigation: launchDefinitionGlobalNavigationItem(),
  },
})
