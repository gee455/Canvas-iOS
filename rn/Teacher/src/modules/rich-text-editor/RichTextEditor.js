//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import React, { Component } from 'react'
import Screen from '../../routing/Screen'
import i18n from 'format-message'
import RCE, { type Props } from '../../common/components/rich-text-editor/RichTextEditor'

type OwnProps = Props & {
  onChangeValue?: (value: string) => void,
}

export default class RichTextEditor extends Component<OwnProps, any> {
  editor: ?RCE

  done = async () => {
    if (!this.editor) return
    const html = await this.editor.getHTML()
    this.props.onChangeValue && this.props.onChangeValue(html)
    this.props.navigator.dismiss()
  }

  render () {
    return (
      <Screen
        showDismissButton={false}
        leftBarButtons={[
          {
            title: i18n('Done'),
            testID: 'rich-text-editor.dismiss',
            style: 'done',
            action: this.done,
          },
        ]}
      >
        <RCE
          {...this.props}
          ref={(r) => { this.editor = r }}
        />
      </Screen>
    )
  }
}
