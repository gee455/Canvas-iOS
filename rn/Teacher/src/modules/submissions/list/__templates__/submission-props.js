//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
