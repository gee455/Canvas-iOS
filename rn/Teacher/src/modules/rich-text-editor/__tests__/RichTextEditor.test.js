//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'
import RichTextEditor from '../RichTextEditor'
import * as template from '../../../__templates__/'

describe('RichTextEditor', () => {
  let props
  beforeEach(() => {
    props = {
      navigator: template.navigator(),
    }
  })

  it('renders', () => {
    expect(shallow(<RichTextEditor {...props} />)).toMatchSnapshot()
  })

  it('passes back html on done pressed', async () => {
    props.onChangeValue = jest.fn()
    const view = shallow(<RichTextEditor {...props} />)
    const html = '<div>HTML ON DONE</div>'
    view.find('RichTextEditor').getElement().ref({ getHTML: jest.fn(() => Promise.resolve(html)) })
    await view.prop('leftBarButtons')[0].action()
    expect(props.onChangeValue).toHaveBeenCalledWith(html)
  })

  it('dismisses on done pressed', async () => {
    props.navigator = template.navigator({ dismiss: jest.fn() })
    const view = shallow(<RichTextEditor {...props} />)
    view.find('RichTextEditor').getElement().ref({ getHTML: jest.fn(() => Promise.resolve('')) })
    await view.prop('leftBarButtons')[0].action()
    expect(props.navigator.dismiss).toHaveBeenCalled()
  })
})
