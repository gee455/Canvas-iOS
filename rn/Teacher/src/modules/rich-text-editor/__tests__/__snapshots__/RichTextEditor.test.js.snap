// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextEditor renders 1`] = `
<Screen
  leftBarButtons={
    Array [
      Object {
        "action": [Function],
        "style": "done",
        "testID": "rich-text-editor.dismiss",
        "title": "Done",
      },
    ]
  }
  navBarStyle="light"
  showDismissButton={false}
  statusBarStyle="default"
>
  <RichTextEditor
    navigator={
      Object {
        "dismiss": [Function],
        "dismissAllModals": [Function],
        "isModal": false,
        "pop": [Function],
        "show": [Function],
        "traitCollection": [Function],
      }
    }
  />
</Screen>
`;
